oauth2_provider/middleware.py

```diff
@@ -52,8 +52,8 @@ def __init__(self, get_response):
 
     def __call__(self, request):
         authheader = request.META.get("HTTP_AUTHORIZATION", "")
-        if authheader.startswith("Bearer"):
-            tokenstring = authheader.split()[1]
+        if authheader.startswith("Bearer") and len(authheader.split(maxsplit=1)) == 2:
+            tokenstring = authheader.split(maxsplit=1)[1]
```

Comment:
I suggest not calling split() twice:
```suggestion
        splits = autheader.split(maxsplit=1)
        if authheader.startswith("Bearer") and len(splits) == 2:
            tokenstring = splits[1]
```

Reply:
Thank you for your suggestion, I will make the changes by today!